lib/lroPollStrategy.ts

```diff
@@ -400,6 +424,17 @@ class AzureAsyncOperationLROPollStrategy extends LROPollStrategy {
 class GetResourceLROPollStrategy extends LROPollStrategy {
   protected sendPollRequest(): Promise<void> {
     return this.getStatus(this._initialRequestUrl).then(result => {
+      const statusCode: number = result.status;
+      const responseBody: any = result.parsedBody;
+      if (statusCode !== 200 && statusCode !== 201 && statusCode !== 202 && statusCode !== 204) {
+        const error = new RestError(`Invalid status code with response body "${result.bodyAsText}" occurred when polling for operation status.`);
+        error.statusCode = statusCode;
+        error.request = stripRequest(result.request);
+        error.response = result;
```

Comment:
Is there stripping that should happen on the response as well?

Reply:
I've thought about this previously, but currently we don't strip the response anywhere. If we find headers that should be removed, then we should add a stripResponse() method, but we don't have one now.